src/hotspot/os/windows/os_windows.cpp

```diff
@@ -1947,7 +1947,10 @@ void os::win32::print_windows_version(outputStream* st) {
       // - 2016 GA 10/2016 build: 14393
       // - 2019 GA 11/2018 build: 17763
       // - 2022 GA 08/2021 build: 20348
-      if (build_number > 20347) {
+      // - 2025 Preview build   : 26040
```

Comment:
Just for the record, Microsoft released Windows Server 2025 on Nov 1st, 2025 and the official build number is 26100.
```
// - 2025 GA 11/2024 build: 26100`
if (build_number > 26099) {
  st->print("Server 2025");
```
Reference: https://learn.microsoft.com/en-us/windows-server/get-started/windows-server-release-info

Reply:
Thanks this is what I got this week too from our Microsoft experts.
I added the info to this JBS bug 8340387.
With the the check present, we detect also some previews which is not bad imho.